### PR TITLE
[BD-6] OEP-18 Compliance

### PR DESCRIPTION
**Pull request by morenol**
In https://github.com/edx/ecommerce-worker/pull/60 I made the changes to upgrade to python3, but I think that there are a lot of changes for a single PR so I want to split these changes in different PRs.

In this PR, 

I make this repo OEP-18 compliant and ran python-modernize

And then move tests to tox and pytest, 

In a future PR, I will add python3 tests and fix them

## Reviewers
- [ ] @awais786 
- [ ] @ericfab179

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @morenol! I've created [OSPR-4553](https://openedx.atlassian.net/browse/OSPR-4553) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Review by UsamaSadiq — APPROVED** [review]

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,28 @@
+[tox]
+envlist = py{27,35,38},quality
```
Comment (mraarif) [inline tox.ini:2]: why are we still adding python 2.7?

File: requirements/app.in [hunk 2]
```diff
@@ -0,0 +1,7 @@
+-c constraints.txt
+
+## Dependencies for the edx-ecommerce-worker standalone app
+
+redis==2.10.6
```
Comment (mraarif) [inline requirements/app.in]: can we just remove the version constraints here and try to constrain (only) the packages (using `constraints.txt`) that fail/break on a newer version.

File: requirements/base.in [hunk 3]
```diff
@@ -0,0 +1,6 @@
+-c constraints.txt
+
+## Dependencies for the edx-ecommerce-worker library
+
+celery>=3.1.18,<4.0.0
```
Comment (mraarif) [inline requirements/base.in]: same here

File: requirements/optional.in [hunk 4]
```diff
@@ -0,0 +1,4 @@
+-c constraints.txt
+
+# Optional packages
+newrelic==4.8.0.110
```
Comment (mraarif) [inline requirements/optional.in]: same here

File: requirements/production.in [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Packages required in a production environment
+-r base.txt
+-c constraints.txt
+
+PyYAML==3.11
```
Comment (mraarif) [inline requirements/production.in]: and here

File: requirements/test.in [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+coverage
+ddt==1.0.1
+edx-lint
+httpretty==0.8.10
+mock==1.3.0
+pytest
+pytest-cov
+pycodestyle
+testfixtures==4.13.3
```
Comment (mraarif) [inline requirements/test.in]: same here

**Review by mzulqarnain1 — APPROVED** [review]

File: tox.ini (hunk 1 above)
Comment (morenol) [reply tox.ini:2]: This project is currently deployed with python2.7, This is the first step in order to make this runs with python 3.5, the next commits are in  https://github.com/edx/ecommerce-worker/pull/60 but I wanted to seperate the PRs in order to be more easy to review the changes.

**Review by ericfab179 — APPROVED** [review]

**Review by mraarif — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@morenol 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/ecommerce-worker&entry.867055334=https://github.com/edx/ecommerce-worker/pull/61&entry.1484655318=https://github.com/morenol&entry.752974735=2020-05-18+17:52&entry.1917517419=2020-05-20+16:49&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**